Check all Trino catalogs if no catalog configured

daniel-cortez-stevenson opened the pull request:
## What type of PR is this? 

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## Description

With a Trino Data Source, if you do not specify a `catalog`, no `schema.table`s appear in the left-hand side of the Redash Query view. The connector will not throw an error when you click 'Test Connection'. We have data from multiple catalogs configured to be queried by Trino. We'd like to query them and join them using a single Redash Data Source. Being able to discover in the Redash Query view which catalogs you have access to is an improvement on the UX in this scenario.

The logic goes like this:
- If a catalog is configured, you get the same result as before this change.
- If a catalog is not configured, query 'SHOW CATALOGS' to get catalogs.
- If a returned catalog has a `.` in its name, then double-quote the catalog name.
- For each catalog, discover schema, table, & columns.
- The table name returned includes the catalog name.

## How is this tested?

- [x] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [x] Manually
- [ ] N/A

Sorry having issues with `docker-compose up --build` and can't test this locally at the moment.

## Related Tickets & Documents
None

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
None

justinclift commented:
To expand that mention of [the wiki](https://github.com/getredash/redash/wiki/Local-development-setup), there are probably two main pages of interest:

1. [Local development setup](https://github.com/getredash/redash/wiki/Local-development-setup) - How to set up an Ubuntu 20.04 desktop, for Redash development purposes
2. [Testing your changes](https://github.com/getredash/redash/wiki/Testing-your-changes) - Follows on from the local dev setup page, explaining how the Redash tests work and how to run them

daniel-cortez-stevenson commented:
Hey @guidopetri LGTM 🚀

justinclift commented:
@daniel-cortez-stevenson Cool, looks like this is pretty close to ready. :smile:

How feasible is it to add some test coverage to this, to make the Codecov check happy?

guidopetri commented:
Well, we'll need some test coverage in order to be able to merge. 😄

daniel-cortez-stevenson commented:
Hey @guidopetri I was able to test this out in the UI locally and fixed a few issues with the feature. Now it works! 

There are multiple catalogs represented in the schema in this image:

<img width="630" alt="image" src="https://github.com/getredash/redash/assets/26047171/7f71bd64-274f-440a-98a7-7b3587c1fac8">

With the settings in this image:

<img width="542" alt="image" src="https://github.com/getredash/redash/assets/26047171/35191e4c-9336-4097-a823-4f6b1981c5bc">

Unfortunately, I can't seem to get `pip install -r requirements_all_ds.txt` working on my Mac OS Ventura 13.5 (M2 chip). I believe that `make test` is not running as a result of that ... when I try to run `python tests/query_runner/test_trino.py` I get the error about `redash` not existing 😭 .

So, I added some unit tests but I haven't actually run them (lol).

daniel-cortez-stevenson commented:
And if you were to set the catalog to `memory` in this instance ...

<img width="300" alt="image" src="https://github.com/getredash/redash/assets/26047171/99c6f71b-20d9-4b1d-8da4-2fae07c9afad">

As expected, you get only `memory` catalog tables without the catalog name.

daniel-cortez-stevenson commented:
@guidopetri Ah, I wasn't thinking and merged the master branch into the PR branch. We prefer rebase, right? I can undo it if that's the case.

guidopetri commented:
It's no big deal, we squash the PR before merging anyhow. :)

As for M2... I feel your pain. If we fix the formatting, the CI should run the tests though.

daniel-cortez-stevenson commented:
@guidopetri `make build` runs fine locally but `make compose_build` is throwing an error:

```
=> ERROR [stage-1  8/11] RUN /etc/poetry/bin/poetry install --only main,all_ds,dev --no-root --no-interaction --no-ansi                                                                                        42.2s 
------                                                                                                                                                                                                                
 > [stage-1  8/11] RUN /etc/poetry/bin/poetry install --only main,all_ds,dev --no-root --no-interaction --no-ansi:                                                                                                    
#0 0.498 Skipping virtualenv creation, as specified in config file.                                                                                                                                                   
#0 0.582 Installing dependencies from lock file                                                                                                                                                                       
#0 1.013                                                                                                                                                                                                              
#0 1.013 Package operations: 220 installs, 1 update, 0 removals                                                                                                                                                       
#0 1.013 
#0 1.013   • Installing pycparser (2.21)
#0 1.014   • Installing six (1.16.0)
#0 1.337   • Installing cffi (1.15.1)
#0 1.338   • Installing exceptiongroup (1.1.3)
#0 1.340   • Installing idna (3.4)
#0 1.341   • Installing markupsafe (2.1.1)
#0 1.342   • Installing packaging (23.1)
#0 1.342   • Installing pyasn1 (0.5.0)
#0 1.344   • Installing jmespath (1.0.1)
#0 1.345   • Installing python-dateutil (2.8.0)
#0 1.346   • Installing urllib3 (1.26.16)
#0 1.347   • Installing sniffio (1.3.0)
#0 1.412   • Installing zipp (3.16.2)
#0 1.655   • Installing anyio (4.0.0)
#0 1.656   • Installing async-timeout (4.0.3)
#0 1.656   • Installing blinker (1.6.2)
#0 1.657   • Installing botocore (1.31.8)
#0 1.657   • Installing cachetools (5.3.1)
#0 1.659   • Installing certifi (2023.7.22)
#0 1.660   • Installing charset-normalizer (3.2.0)
#0 1.661   • Installing click (8.1.3)
#0 1.662   • Installing cryptography (41.0.3)
#0 1.663   • Installing h11 (0.14.0)
#0 1.720   • Installing importlib-metadata (6.8.0)
#0 1.725   • Installing itsdangerous (2.1.2)
#0 1.729   • Installing jinja2 (3.1.2)
#0 1.759   • Installing marshmallow (3.20.1)
#0 1.792   • Installing mdurl (0.1.2)
#0 1.801   • Installing pyasn1-modules (0.3.0)
#0 1.812   • Installing pyparsing (2.4.7)
#0 1.819   • Installing python-dotenv (0.19.2)
#0 1.866   • Installing rsa (4.9)
#0 1.894   • Installing typing-extensions (4.7.1)
#0 1.982   • Installing werkzeug (2.3.6)
#0 2.015   • Installing wrapt (1.15.0)
#0 4.497   • Installing asn1crypto (1.5.1)
#0 4.498   • Installing bcrypt (4.0.1)
#0 4.498   • Installing deprecated (1.2.14)
#0 4.498   • Installing distlib (0.3.7)
#0 4.499   • Installing elementpath (4.1.5)
#0 4.500   • Installing environs (9.5.0)
#0 4.501   • Installing filelock (3.12.3)
#0 4.502   • Installing flask (2.3.2)
#0 4.503   • Installing google-auth (2.22.0)
#0 4.504   • Installing httpcore (0.16.3)
#0 4.576   • Installing httplib2 (0.19.0)
#0 4.605   • Installing importlib-resources (6.0.1)
#0 4.638   • Installing iniconfig (2.0.0)
#0 4.646   • Installing isodate (0.6.1)
#0 4.727   • Installing mako (1.2.4)
#0 4.735   • Installing markdown-it-py (3.0.0)
#0 4.743   • Installing mysql-connector (2.2.9)
#0 4.758   • Installing parso (0.8.3)
#0 4.812   • Installing platformdirs (3.8.1)
#0 4.819   • Installing pluggy (1.3.0)
#0 4.833   • Installing ply (3.11)
#0 4.837   • Installing pygments (2.16.1)
#0 4.858   • Installing pyjwt (2.4.0)
#0 4.864   • Installing pynacl (1.5.0)
#0 4.876   • Installing pyopenssl (23.2.0)
#0 4.888   • Installing pytz (2023.3.post1)
#0 4.903   • Installing redis (4.6.0)
#0 4.926   • Installing requests (2.31.0)
#0 5.123   • Installing rfc3986 (1.5.0)
#0 5.189   • Installing s3transfer (0.6.2)
#0 5.219   • Installing scramp (1.1.0)
#0 5.249   • Updating setuptools (57.5.0 -> 68.2.0)
#0 5.298   • Installing sqlalchemy (1.3.24)
#0 5.491   • Installing tomli (2.0.1)
#0 5.570   • Installing wcwidth (0.2.6)
#0 8.888   • Installing adal (1.2.7)
#0 8.889   • Installing alembic (1.12.0)
#0 8.890   • Installing aniso8601 (8.0.0)
#0 8.890   • Installing attrs (23.1.0)
#0 8.891   • Installing appdirs (1.4.4)
#0 8.891   • Installing bitarray (2.8.1)
#0 8.892   • Installing boto (2.49.0)
#0 8.895   • Installing boto3 (1.28.8)
#0 8.895   • Installing backports-zoneinfo (0.2.1)
#0 8.896   • Installing cfgv (3.4.0)
#0 8.984   • Installing ciso8601 (2.3.0)
#0 8.985   • Installing coverage (7.2.7)
#0 9.002   • Installing croniter (1.4.1)
#0 9.065   • Installing databend-py (0.4.6)
#0 9.099   • Installing defusedxml (0.7.1)
#0 9.110   • Installing dnspython (2.4.2)
#0 9.138   • Installing et-xmlfile (1.1.0)
#0 9.173   • Installing flask-sqlalchemy (2.5.1)
#0 9.264   • Installing future (0.18.3)
#0 9.272   • Installing futures (3.0.5)
#0 9.364   • Installing geomet (0.1.2)
#0 9.392   • Installing google-auth-httplib2 (0.1.0)
#0 9.464   • Installing greenlet (1.1.3)
#0 9.500   • Installing grpcio (1.58.0)
#0 9.516   • Installing httpx (0.23.3)
#0 9.820   • Installing identify (2.5.27)
#0 9.991   • Installing inflection (0.3.1)
#0 10.09   • Installing jedi (0.19.0)
#0 10.43   • Installing limits (3.6.0)
#0 10.75   • Installing maxminddb (2.4.0)
#0 10.94   • Installing msgpack (1.0.5)
#0 11.23   • Installing mysqlclient (2.1.1)
#0 11.24   • Installing ndg-httpsclient (0.5.1)
#0 11.28   • Installing netifaces (0.11.0)
#0 11.43   • Installing nodeenv (1.8.0)
#0 11.44   • Installing numpy (1.24.4)
#0 11.44   • Installing nzpy (1.15)
#0 11.53   • Installing ordered-set (4.1.0)
#0 11.58   • Installing oscrypto (1.3.0)
#0 11.73   • Installing paramiko (3.3.1)
#0 11.80   • Installing prompt-toolkit (3.0.39)
#0 11.82   • Installing protobuf (3.18.3)
#0 12.14   • Installing psutil (5.9.5)
#0 12.29   • Installing pure-sasl (0.6.2)
#0 12.55   • Installing pycryptodomex (3.18.0)
#0 13.28   • Installing pyrsistent (0.19.3)
#0 13.33   • Installing pytest (7.4.0)
#0 13.59   • Installing pyyaml (6.0.1)
#0 13.83   • Installing rdflib (6.3.2)
#0 14.07   • Installing requests-toolbelt (1.0.0)
#0 14.17   • Installing rich (13.5.2)
#0 14.20   • Installing rq (1.9.0)
#0 14.23   • Installing simplejson (3.16.0)
#0 14.35   • Installing sortedcontainers (2.4.0)
#0 14.55   • Installing sqlalchemy-utils (0.34.2)
#0 14.58   • Installing tenacity (8.2.3)
#0 14.59   • Installing thrift (0.16.0)
#0 14.69   • Installing thriftpy2 (0.4.16)
#0 14.71   • Installing tomlkit (0.12.1)
#0 14.73   • Installing tzlocal (2.1)
#0 14.85   • Installing ua-parser (0.18.0)
#0 14.96   • Installing uritemplate (3.0.1)
#0 14.98   • Installing validators (0.22.0)
#0 15.08   • Installing virtualenv (20.24.1)
#0 15.22   • Installing websocket-client (1.6.2)
#0 15.24   • Installing wraptor (0.7.0)
#0 15.46   • Installing wtforms (2.2.1)
#0 15.60   • Installing xmlschema (2.4.0)
#0 16.14   • Installing zope-event (5.0)
#0 16.29   • Installing zope-interface (6.0)
#0 21.66   • Installing advocate (1.0.0)
#0 21.66   • Installing atsd-client (3.0.5)
#0 21.66   • Installing authlib (0.15.5)
#0 21.66   • Installing cassandra-driver (3.21.0)
#0 21.66   • Installing azure-kusto-data (0.0.35)
#0 21.66   • Installing cmem-cmempy (21.2.3)
#0 21.66   • Installing databend-sqlalchemy (0.2.4)
#0 21.67   • Installing disposable-email-domains (0.0.95)
#0 21.67   • Installing flask-limiter (3.3.1)
#0 21.67   • Installing flask-login (0.6.0)
#0 21.88   • Installing flask-mail (0.9.1)
#0 21.90   • Installing flask-migrate (2.5.2)
#0 21.90   • Installing flask-restful (0.3.10)
#0 21.91   • Installing flask-talisman (0.7.0)
#0 21.92   • Installing flask-wtf (1.1.1)
#0 21.93   • Installing freezegun (1.2.1)
#0 21.96   • Installing funcy (1.13)
#0 21.99   • Installing gevent (22.10.1)
#0 22.01   • Installing google-api-python-client (1.7.11)
#0 22.02   • Installing gspread (3.1.0)
#0 22.06   • Installing gunicorn (20.0.4)
#0 22.09   • Installing impyla (0.16.0)
#0 22.10   • Installing influxdb (5.2.3)
#0 22.25   • Installing jsonschema (3.1.1)
#0 22.33   • Installing jwcrypto (1.5.0)
#0 22.34   • Installing maxminddb-geolite2 (2018.703)
#0 22.35   • Installing memsql (3.2.0)
#0 22.53   • Installing mock (5.0.2)
#0 22.54   • Installing nzalchemy (11.0.2)
#0 22.85   • Installing oauth2client (4.1.3)
#0 22.90   • Installing openpyxl (3.0.7)
#0 23.24   • Installing oracledb (1.4.0)
#0 23.29   • Installing pandas (1.3.4)
#0 23.77   • Installing parsedatetime (2.4)
#0 23.93   • Installing passlib (1.7.3)
#0 24.20   • Installing phoenixdb (0.7)
#0 24.21   • Installing pinotdb (5.1.0)
#0 24.25   • Installing pre-commit (3.3.3)
#0 24.29   • Installing psycopg2-binary (2.9.6)
#0 24.42   • Installing ptpython (3.0.23)
#0 24.43   • Installing ptvsd (4.3.2)
#0 24.45   • Installing pyathena (1.11.5)
#0 24.67   • Installing pydgraph (2.0.2)
#0 24.73   • Installing pydruid (0.5.7)
#0 24.74   • Installing pyexasol (0.12.0)
#0 24.80   • Installing pygridgain (1.4.0)
#0 24.90   • Installing pyhive (0.6.1)
#0 24.98   • Installing pyignite (0.6.1)
#0 25.12   • Installing pymongo (4.3.3)
#0 26.01   • Installing pymssql (2.2.8)
#0 26.07   • Installing pyodbc (4.0.28)
#0 26.23   • Installing pypd (1.1.0)
#0 26.54   • Installing pysaml2 (7.3.1)
#0 27.17   • Installing pystache (0.6.0)
#0 27.63   • Installing pytest-cov (4.1.0)
#0 27.92   • Installing python-arango (6.1.0)
#0 28.18   • Installing python-rapidjson (1.1)
#0 28.38   • Installing qds-sdk (1.16.1)
#0 28.54   • Installing regex (2023.8.8)
#0 28.59   • Installing requests-aws-sign (0.1.5)
#0 28.84   • Installing restrictedpython (6.2)
#0 29.21   • Installing rq-scheduler (0.11.0)
#0 29.35   • Installing sasl (0.3.1)
#0 29.46   • Installing semver (2.8.1)
#0 29.55   • Installing sentry-sdk (1.28.1)
#0 29.82   • Installing simple-salesforce (0.74.3)
#0 30.10   • Installing snowflake-connector-python (3.1.0)
#0 30.10   • Installing sqlalchemy-searchable (1.2.0)
#0 30.12   • Installing sqlparse (0.4.4)
#0 30.28   • Installing sshtunnel (0.1.5)
#0 30.60   • Installing statsd (3.3.0)
#0 30.83   • Installing supervisor (4.1.0)
#0 31.03   • Installing supervisor-checks (0.8.1)
#0 31.53   • Installing td-client (1.0.0)
#0 31.83   • Installing thrift-sasl (0.4.3)
#0 31.93   • Installing trino (0.326.0)
#0 32.03   • Installing user-agents (2.0)
#0 32.04   • Installing vertica-python (1.1.1)
#0 32.10   • Installing watchdog (3.0.0)
#0 32.14   • Installing xlrd (2.0.1)
#0 32.28   • Installing xlsxwriter (1.2.2)
#0 41.78 
#0 41.78   ChefBuildError
#0 41.78 
#0 41.78   Backend subprocess exited when trying to invoke build_wheel
#0 41.78   
#0 41.78   setup.py: platform.system() => Linux
#0 41.78   setup.py: platform.architecture() => ('64bit', '')
#0 41.78   setup.py: platform.libc_ver() => ('glibc', '2.28')
#0 41.78   setup.py: include_dirs => []
#0 41.78   setup.py: library_dirs => []
#0 41.78   running bdist_wheel
#0 41.78   running build
#0 41.78   running build_py
#0 41.78   creating build
#0 41.78   creating build/lib.linux-aarch64-cpython-38
#0 41.78   creating build/lib.linux-aarch64-cpython-38/pymssql
#0 41.78   copying src/pymssql/__init__.py -> build/lib.linux-aarch64-cpython-38/pymssql
#0 41.78   running build_ext
#0 41.78   Compiling src/pymssql/_mssql.pyx because it changed.
#0 41.78   [1/1] Cythonizing src/pymssql/_mssql.pyx
#0 41.78   building 'pymssql._mssql' extension
#0 41.78   creating build/temp.linux-aarch64-cpython-38
#0 41.78   creating build/temp.linux-aarch64-cpython-38/src
#0 41.78   creating build/temp.linux-aarch64-cpython-38/src/pymssql
#0 41.78   gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/tmp/tmp5c3hrq56/.venv/include -I/usr/local/include/python3.8 -c src/pymssql/_mssql.c -o build/temp.linux-aarch64-cpython-38/src/pymssql/_mssql.o -DMSDBLIB
#0 41.78   gcc -pthread -shared build/temp.linux-aarch64-cpython-38/src/pymssql/_mssql.o -L/usr/local/lib -lsybdb -lgssapi_krb5 -lkrb5 -lssl -lcrypto -o build/lib.linux-aarch64-cpython-38/pymssql/_mssql.cpython-38-aarch64-linux-gnu.so
#0 41.78   /usr/bin/ld: cannot find -lgssapi_krb5
#0 41.78   /usr/bin/ld: cannot find -lkrb5
#0 41.78   collect2: error: ld returned 1 exit status
#0 41.78   warning: src/pymssql/_mssql.pyx:27:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:28:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:29:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:30:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:31:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:32:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:35:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:36:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:38:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:82:0: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:230:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:264:4: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:333:4: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:831:8: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:919:8: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:1005:12: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:1379:16: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   warning: src/pymssql/_mssql.pyx:1713:8: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
#0 41.78   error: command '/usr/bin/gcc' failed with exit code 1
#0 41.78   
#0 41.78 
#0 41.78   at /etc/poetry/venv/lib/python3.8/site-packages/poetry/installation/chef.py:147 in _prepare
#0 41.79       143│ 
#0 41.79       144│                 error = ChefBuildError("\n\n".join(message_parts))
#0 41.79       145│ 
#0 41.79       146│             if error is not None:
#0 41.79     → 147│                 raise error from None
#0 41.79       148│ 
#0 41.79       149│             return path
#0 41.79       150│ 
#0 41.79       151│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:
#0 41.79 
#0 41.79 Note: This error originates from the build backend, and is likely not a problem with poetry but with pymssql (2.2.8) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "pymssql (==2.2.8)"'.
#0 41.79 
------
failed to solve: process "/bin/sh -c /etc/poetry/bin/poetry install --only $install_groups $POETRY_OPTIONS" did not complete successfully: exit code: 1
make: *** [compose_build] Error 17
```

So I'm not able to confirm that the merged in changes to the Trino query_runner (visualizing the column type) are working with this PR 🤔

guidopetri commented:
@daniel-cortez-stevenson Interesting, it looks like from these lines

> #0 41.78   /usr/bin/ld: cannot find -lgssapi_krb5
> #0 41.78   /usr/bin/ld: cannot find -lkrb5

that the arm64 version of the docker image requires [libkrb5-dev](https://packages.ubuntu.com/search?keywords=libkrb5-dev) to be installed. I don't think I had an issue running on amd64 but I'll give it another shot - in the meantime, I approved the workflow run so we can see if the tests pass.

daniel-cortez-stevenson commented:
@guidopetri I managed to get redash installed for dev and fixed the existing test (& added a second test for when a catalog is set). They pass locally. 

I would be very open to any suggestions on the test definitions!

Running make compose build now after I did a `brew install krb5` to check out the viz of column type...

guidopetri commented on tests/query_runner/test_trino.py:
```diff
@@ -0,0 +1,59 @@
+    def side_effect_run_query(query, user):
+        catalog_query_pattern = r"\s*SHOW\s+CATALOGS\s*"
+        schema_query_pattern = r"""
+        \s*SELECT\s+table_schema,\s+table_name,\s+column_name,\s+data_type
+        \s+FROM\s+(\w+\.)?information_schema\.columns
+        \s+WHERE\s+table_schema\s+NOT\s+IN\s+\('pg_catalog',\s+'information_schema'\)\s*
```
I think that instead of hardcoding the queries that get run in the runner here, we could patch the `.run_query` method to return the results in L27 / L29-31 directly. That way we don't depend on the exact implementation of the Trino runner, but rather just on the input/output being exactly the same every time.

daniel-cortez-stevenson replied:
It is addressed in 33d30c6. What do you think about that change @guidopetri ?

daniel-cortez-stevenson commented:
ok! LGTM 🏎️

guidopetri commented on redash/query_runner/trino.py:
```diff
@@ -83,26 +83,48 @@ def type(cls):
+                catalog = row["Catalog"]
+                if "." in catalog:
+                    catalog = f'"{catalog}"'
+                catalog_prefixes.append(f"{catalog}.")
```
The catalog prefix having a `.` suffix looks weird to me, but I don't know how to make the pattern better. Is there a way for the provided catalog to be picked up instead of setting the prefixes to `[""]`?

daniel-cortez-stevenson replied:
Fixed in c682e01

guidopetri commented:
This looks great, thanks @daniel-cortez-stevenson !

Leelongqi commented:
can you tell me your redash vsesion? my install my redash is not trino datasource,https://github.com/getredash/redash/discussions/6477
